### PR TITLE
fix hash aliases

### LINKED ISSUE #334: About the new change in AliasPlugin.js
last week, your team update the version of the enhanced-resolve
however, using v5.9.2, my project cannot build at all, it will be stuck while building. 
It makes me really puzzled, because my project's alias is `'#': path.resolve(__dirname, './src'),`
when I change it into `'balabala': path.resolve(__dirname, './src'),`, it will continue to build. balabala can be anything but #
and when I change the line 47 into "request.path",  `'#': path.resolve(__dirname, './src'),` can also make it.
So I'm wondering the reason why your team make such a change

### PR DESCRIPTION
**Pull request by vankop**
fixes #334

### CODE REVIEW

File: test/alias.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ describe("alias", function () {
 				recursive: "recursive/dir",
 				"/d/dir": "/c/dir",
 				"/d/index.js": "/c/index",
+				// alias configuration should work
+				"#": "/c/dir",
+				"@": "/c/dir",
```
Comment (sokra) [inline test/alias.js:44]: maybe also add a test case below for that

Comment (vankop) [reply test/alias.js:44]: technically it fails without applying.. (just setting crash resolving) lets add usage anyway.

File: lib/AliasPlugin.js [hunk 2]
```diff
@@ -30,6 +31,25 @@ module.exports = class AliasPlugin {
 	 */
 	apply(resolver) {
 		const target = resolver.ensureHook(this.target);
+		const cache = {};
```
Comment (sokra) [inline lib/AliasPlugin.js:34]: The cache looks like an overkill for me here... Does it really make sense here?

**Review by sokra — CHANGES_REQUESTED** [review]

File: lib/AliasPlugin.js (hunk 2 above)
Comment (vankop) [reply lib/AliasPlugin.js:34]: hm.. since mostly we won't have absolute aliases on average algorithm complexity will be `O(1)`.. lets remove it.

**Review by sokra — APPROVED** [review]

**Pull request merged**